### DISCUSSION TITLE
Getting parameter cannot be found that matches AllowBlobPublicAccess error....

### DISCUSSION THREAD

**Discussion by klabranche** [actions/virtual-environments#5712] [General questions]
Following the doc https://github.com/actions/virtual-environments/blob/main/docs/create-image-and-azure-resources.md

On the Import GenerateResourcesAndImage script from /helpers folder, and run GenerateResourcesAndImage function via Powershell.

I get: A parameter cannot be found that matches parameter name 'AllowBlobPublicAccess'

Creates the resource group and deletes it just fine.  

Repo pulled today
Windows 10 21H2
PS - 5.1.19041.1682
Az Module - 8.0.0
Azure CLI - 2.37.0

Any ideas?

**Comment by al-cheb** [comment]
Hey @klabranche,
Please provide a command line how you run the script.

**Reply by klabranche** [reply]
GenerateResourcesAndImage -SubscriptionId myId `
-ResourceGroupName "w2022BAImg-rg" -ImageGenerationRepositoryRoot "$pwd" `
-ImageType Windows2022 -AzureLocation "West US" -EnableHttpsTrafficOnly $true `
-tags @{dept="devops";env="prod"}

![1](https://user-images.githubusercontent.com/1045379/172865425-162a1ac5-6322-4813-b11f-a89b9c03d7e0.png)
![2](https://user-images.githubusercontent.com/1045379/172865462-70e42673-328f-478a-ab4d-852d00f849f0.png)

**Comment by al-cheb** [comment]
I was not able to reproduce with the command line: 
```
GenerateResourcesAndImage -SubscriptionId myId -ResourceGroupName "w2022BAImg-rg" -ImageGenerationRepositoryRoot "$pwd"
-ImageType Windows2022 -AzureLocation "West US" -EnableHttpsTrafficOnly $true -tags @{dept="devops";env="prod"}
```

**Reply by klabranche** [reply]
It definitely has been interesting.  I force installed, allow clobber.  I removed the parameter from the script and from New-AzStorageAccount.  I also tried passing it in... Same error every time.  Has me befuddled atm.

**Reply by klabranche** [reply]
When I run New-AzStorageAccount -ResourceGroupName "w2022BAImg-rg" -AccountName "w2022baimg001" -Location "West US" -SkuName "Standard_LRS" -EnableHttpsTrafficOnly $False 

It works but when I add -AllowBlobPublicAccess I get the same error.  

It's as if I'm using a version of the call different than v8.0 but I thought that would be remedied with force install and allow clobber and I only have v8.0 when I check Find-Module Az

**Reply by al-cheb** [reply]
Try to update Az module to version 7.5.0:
`Install-Module -Name Az -RequiredVersion 7.5.0 -Scope AllUsers -SkipPublisherCheck -Force`

**Reply by klabranche** [reply]
Most interesting.  I ended up force uninstalling ALL versions of the Az module.  Then reinstalled with 7.5.0 as above.  I still get the same error.  Something must be amiss about my box with these modules.....  I just haven't found it, nothing obvious has appeared.

**Reply by klabranche** [reply]
Finally was able to resolve with some old modules removal....

**Comment by samus-aran** [comment]
We're locking this discussion because it has not had recent activity and/or other members have asked for more information to assist you but received no response. Thank you for helping us maintain a productive and tidy community for all our members.
